aiidalab/app.py

```diff
@@ -429,6 +455,52 @@ def _has_git_repo(self):
         except NotGitRepository:
             return False
 
+    @property
+    def environment(self):
+        """Return the environment instance for this app."""
+        return AppEnvironment(self.name)
```

Comment:
Should this return an in-memory `self._environment` attribute instead so that an `AppEnvironment()` object is only instantiated once per `self.name`?

Reply:
Done.

Reply:
I am not sure, but I believe it was done this way because the environment might change, so to be more interactive we might want to load it every time. What do you think @csadorf ?

Reply:
No, the app environment is fully defined through the app name which is immutable by definition. So there is no need for multiple instances. I've committed that micro optimization.

Reply:
alright then!